src/components/Item.jsx

```diff
@@ -0,0 +1,35 @@
+import { useState } from 'react';
+// import itemReducer from '../views/Shopping';
+
+export default function Item({ item, onDeleteItem, onChangeItem }) {
+  const [edit, setEdit] = useState(false);
+  let itemContent;
+
+  if (edit) {
+    itemContent = (
+      <>
+        <input
+          defaultValue={item.text}
+          onChange={(e) => {
+            onChangeItem({ ...item, text: e.target.value });
+          }}
+        />
+        <button onClick={() => setEdit(false)}>save</button>
+      </>
+    );
+  } else {
+    itemContent = (
+      <>
+        <span>{item.text}</span>
+        <button onClick={() => setEdit(true)}>edit</button>
+      </>
+    );
+  }
+  return (
+    <>
+      <input type="checkbox"></input>
```

Comment:
We might want to control the state of this input and tie the `checked `property to the `done ` property of your items in state